src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -4070,5 +4071,136 @@ static void Main()
             Assert.Equal("System.Int32", typeInfo.ConvertedType.ToTestDisplayString());
             Assert.Equal("System.Int32", typeInfo.Type.ToTestDisplayString());
         }
+
+        [Fact]
+        public void TestInOverloadWithIllegalConversion()
+        {
+            var source = @"
+class C
+{
+    public static void Main()
+    {
+        M(new());
+        M(array: new());
+    }
+    static void M(int[] array) { }
+    static void M(int i) { }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,9): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(int[])' and 'C.M(int)'
+                //         M(new());
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(int[])", "C.M(int)").WithLocation(6, 9),
+                // (7,18): error CS8752: The type 'int[]' may not be used as the target type of new()
+                //         M(array: new());
+                Diagnostic(ErrorCode.ERR_TypelessNewIllegalTargetType, "new()").WithArguments("int[]").WithLocation(7, 18)
+                );
+        }
+
+        [Fact]
+        public void TestInOverloadWithUseSiteError()
+        {
+            var missing = @"public class Missing { }";
+            var missingComp = CreateCompilation(missing, assemblyName: "missing");
+
+            var lib = @"
+public class C
+{
+    public void M(Missing m) { }
+    public void M(C c) { }
+}";
+            var libComp = CreateCompilation(lib, references: new[] { missingComp.EmitToImageReference() });
+
+            var source = @"
+class D
+{
+    public void M2(C c)
+    {
+        c.M(new());
```

Comment:
>new() [](start = 12, length = 5)

I think the interesting scenario is to test use-site diagnostics coming from the ```new```, i.e. from the set of available constructors rather than form the set of available ```M```s.

Reply:
Sneaking test in another PR: https://github.com/dotnet/roslyn/pull/43467/commits/61ebdde6ffa275d1201f846c7d676220ff477a05